openshift/dynamic/client.py

```diff
@@ -185,10 +185,15 @@ def patch(self, resource, body=None, name=None, namespace=None, **kwargs):
         if resource.namespaced:
             namespace = self.ensure_namespace(resource, namespace, body)
 
+        merge_type = kwargs.pop('merge_type', None)
         path = resource.path(name=name, namespace=namespace, **kwargs)
 
-        content_type = self.client.\
-            select_header_content_type(['application/json-patch+json', 'application/merge-patch+json', 'application/strategic-merge-patch+json'])
+        if merge_type:
```

Comment:
Why not just make the `content_type` overrideable?

Reply:
Yeah, I think in this client we should make the content_type overrideable, but Ansible side should expose the simpler merge type.

Reply:
@djzager that's fine with me - I'll change merge_type to content_type and handle the merge_type on the Ansible side.